drivers/serial/pty.c

```diff
@@ -583,31 +515,13 @@ static ssize_t pty_read(FAR struct file *filep, FAR char *buffer, size_t len)
 
           if (ch != '\r' || (dev->pd_iflag & IGNCR) == 0)
             {
-              /* Transfer the (possibly translated) character and update the
-               * count of bytes transferred.
-               */
-
-              *buffer++ = ch;
-              ntotal++;
+              buffer[j++] = ch;
             }
         }
-    }
-  else
-#endif
-    {
-      /* NOTE: the source pipe will block if no data is available in
-       * the pipe.   Otherwise, it will return data from the pipe.  If
-       * there are fewer than 'len' bytes in the, it will return with
-       * ntotal < len.
-       *
-       * REVISIT: Should not block if the oflags include O_NONBLOCK.
-       * How would we ripple the O_NONBLOCK characteristic to the
-       * contained source pipe? file_fcntl()?  Or FIONREAD?  See the
-       * TODO comment at the top of this file.
-       */
 
-      ntotal = file_read(&dev->pd_src, buffer, len);
+      ntotal = j;
```

Comment:
@xiaoxiang781216 do we need to read the reset of data if `len != j`?

Reply:
read allow to return the size smaller than the buffer size even there is more input available in the internal soft/hardware buffer per the spec. But, zero need handle specially since it indicate the end of stream.
The update handle this case now, please review again.